lib/openhab/core/emulate_hash.rb

```diff
@@ -0,0 +1,233 @@
+                     :to_a,
+                     :to_h,
+                     :transform_keys,
+                     :transform_values,
+                     :values,
+                     :value?
+
+      # @!visibility private
+      def replace
+        raise NotImplementedError
+      end
+
+      # @!visibility private
+      def to_hash
+        raise NotImplementedError
+      end
+
+      # @!visibility private
+      def fetch(key)
+        raise NotImplementedError
+      end
+
+      # @!visibility private
+      def store(key, value)
+        raise NotImplementedError
+      end
+
+      # @!visibility private
+      def [](key)
+        fetch(key, nil)
+      end
+
+      # @!visibility private
+      def []=(key, value)
+        store(key, value)
+      end
+
+      # @!visibility private
+      def assoc(key)
+        to_hash.assoc(key.to_s)
+      end
+
+      # @!visibility private
+      def clear
+        replace({})
+      end
+
+      # @!visibility private
+      def merge!(*others, &block)
+        return self if others.empty?
+
+        new_config = to_h
+        others.each do |h|
+          new_config.merge!(h.transform_keys(&:to_s), &block)
+        end
+        replace(new_config)
+      end
+      alias_method :update, :merge!
+
+      # @!visibility private
+      def dig(*keys)
+        to_hash.dig(*keys.map(&:to_s))
+      end
+
+      # @!visibility private
+      def compact!
+        replace(compact)
+      end
+
+      # @!visibility private
+      def slice(*keys)
+        to_h.slice(*keys.map(&:to_s))
+      end
+
+      # @!visibility private
+      def delete_if(&block)
+        raise NotImplementedError unless block
+
+        replace(to_h.delete_if(block))
+      end
+
+      # @!visibility private
+      def keep_if(&block)
+        select!(&block)
+        self
+      end
+
+      # @!visibility private
+      def except(*keys)
+        to_hash.except(*keys.map(&:to_s))
+      end
+
+      # @!visibility private
+      def select!(&block)
+        raise NotImplementedError unless block
+
+        r = to_hash.select!(&block)
+        replace(r) if r
+      end
+      alias_method :filter!, :select!
+
+      # @!visibility private
+      def reject!(&block)
+        raise NotImplementedError unless block
+
+        r = to_hash.reject!(&block)
+        replace(r) if r
+      end
+
+      # @!visibility private
+      def fetch_values(*keys, &block)
+        to_hash.fetch_values(*keys.map(&:to_s), &block)
+      end
+
+      # @!visibility private
+      def transform_keys!(*args, &block)
+        replace(transform_keys(*args, &block))
+      end
+
+      # @!visibility private
+      def transform_values!(&block)
+        raise NotImplementedError unless block
+
+        replace(transform_values(&block))
+      end
+
+      # @!visibility private
+      def values_at(*keys)
+        to_hash.values_at(*keys.map(&:to_s))
+      end
+
+      # @!visibility private
+      def <(other)
```

Comment:
all of these comparisons should delegate directly to `to_map` at the top, without stringifying keys, for the same reason as the comparison in Configuration. also, you shouldn't call `to_hash` on `other` without checking if it responds to it (but delegating instead of re-implementing will get you that as well)